GDAPI/GDAPI/Objects/General/BaseCryptoActionResult.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace GDAPI.Objects.General
+{
+    /// <summary>The base class to support objects containing crypto action results.</summary>
+    public class BaseCryptoActionResult
```

Comment:
We might want to make this abstract and implement the Decryption Result as an abstract method.

Reply:
I forgot to make this abstract, and the decryption/encryption results should share that field, since they both have the same functionality; it's just the name that changes.

Reply:
it probably should be called `CipherResult`

Reply:
True, other operations could be encapsulated in such a manner.